packages/block-library/src/post-author/index.php

```diff
@@ -5,18 +5,136 @@
+ */
+function post_author_build_css_font_sizes( $attributes ) {
+	// CSS classes.
+	$font_sizes = array(
+		'css_classes'   => array(),
+		'inline_styles' => '',
+	);
+
+	$has_named_font_size  = array_key_exists( 'fontSize', $attributes );
+	$has_custom_font_size = array_key_exists( 'customFontSize', $attributes );
+
+	if ( $has_named_font_size ) {
+		// Add the font size class.
+		$font_sizes['css_classes'][] = sprintf( 'has-%s-font-size', $attributes['fontSize'] );
+	} elseif ( $has_custom_font_size ) {
+		// Add the custom font size inline style.
+		$font_sizes['inline_styles'] = sprintf( 'font-size: %spx;', $attributes['customFontSize'] );
+	}
+
+	return $font_sizes;
+}
+
 /**
  * Renders the `core/post-author` block on the server.
  *
- * @return string Returns the filtered post author for the current post wrapped inside "h6" tags.
+ * @param  Object $block The block to render.
+ * @return string Returns the rendered author block.
  */
-function render_block_core_post_author() {
+function render_block_core_post_author( $block ) {
+
+	$attributes = $block->attributes;
+
+	if ( empty( $attributes ) ) {
+		return '';
+	}
+
 	$post = gutenberg_get_post_from_context();
+
 	if ( ! $post ) {
 		return '';
 	}
-	// translators: %s: The author.
-	return '<address>' . sprintf( __( 'By %s' ), get_the_author() ) . '</address>';
+
+	$avatar = ! empty( $attributes['avatarSize'] ) ? get_avatar(
+		$post->post_author,
+		$attributes['avatarSize']
+	) : null;
+
+	$byline     = ! empty( $attributes['byline'] ) ? $attributes['byline'] : false;
+	$colors     = post_author_build_css_colors( $attributes );
+	$font_sizes = post_author_build_css_font_sizes( $attributes );
+	$classes    = array_merge(
+		$colors['css_classes'],
+		$font_sizes['css_classes'],
+		array( 'wp-block-post-author' ),
+		isset( $attributes['className'] ) ? array( $attributes['className'] ) : array(),
+		isset( $attributes['itemsJustification'] ) ? array( 'items-justified-' . $attributes['itemsJustification'] ) : array(),
+		isset( $attributes['align'] ) ? array( 'align' . $attributes['align'] ) : array()
+	);
+
+	$class_attribute = sprintf( ' class="%s"', esc_attr( implode( ' ', $classes ) ) );
+	$style_attribute = ( $colors['inline_styles'] || $font_sizes['inline_styles'] )
+		? sprintf( ' style="%s"', esc_attr( $colors['inline_styles'] ) . esc_attr( $font_sizes['inline_styles'] ) )
+		: '';
+
+	return sprintf( '<div %1$s %2$s>', $class_attribute, $style_attribute ) .
+		( $attributes['showAvatar'] ? '<div class="wp-block-post-author__avatar">' . $avatar . '</div>' : '' ) .
+		'<div class="wp-block-post-author__content">' .
+			( ! empty( $byline ) ? '<p class="wp-block-post-author__byline">' . $byline . '</p>' : '' ) .
+			'<p class="wp-block-post-author__name">' . get_the_author_meta( 'display_name' ) . '</p>' .
+			( ! empty( $attributes['showBio'] ) ? '<p class="wp-block-post-author__bio">' . get_the_author_meta( 'user_description' ) . '</p>' : '' ) .
```

Comment:
These template tags implicitly rely on the `$post` global when not given a second argument, which isn't (shouldn't) be the same as the `$post` in scope here. Just as in the above `get_avatar` call passing `$post->post_author`, we should provide this as the second argument `$user_id` here.

```suggestion
			'<p class="wp-block-post-author__name">' . get_the_author_meta( 'display_name', $post->post_author ) . '</p>' .
			( ! empty( $attributes['showBio'] ) ? '<p class="wp-block-post-author__bio">' . get_the_author_meta( 'user_description', $post->post_author ) . '</p>' : '' ) .
```

Reference: https://developer.wordpress.org/reference/functions/get_the_author_meta/

Depending on timing, this may change as a result of revisions in #21696, which I'm currently working to rebase and which led me to this (since it breaks in converting to use block context).

Reply:
Thank you for fixing this.